integration-tests/engine-hdb/src/test/java/com/sap/xsk/hdb/ds/itest/hdbdd/XSKHDBDDHanaITTest.java

```diff
@@ -33,89 +33,92 @@
+    facade.clearCache();
+  }
+
+  @Test
+  public void testHDBDDWithManagedAssOnDiffSchema() throws Exception {
+    try (Connection connection = datasource.getConnection(); Statement stmt = connection.createStatement()) {
+      try {
+        HanaITestUtils.createSchema(stmt, TEST_SCHEMA);
+
+        LocalResource resource = XSKHDBTestModule.getResources( //
+            "/usr/local/target/dirigible/repository/root", //
+            "/registry/public/itest/ProductsWithManagedAssItest.hdbdd", //
+            "/registry/public/itest/ProductsWithManagedAssItest.hdbdd" //
+        );
+
+        facade.handleResourceSynchronization(resource);
+        facade.updateEntities();
+
+        assertTrue(HanaITestUtils.checkExistOfTable(connection, "itest::ProductsWithManagedAssItest.Orders", TEST_SCHEMA));
+        assertTrue(HanaITestUtils.checkExistOfPublicSynonym(connection, "itest::ProductsWithManagedAssItest.Orders"));
+
+        assertTrue(HanaITestUtils.checkExistOfTable(connection, "itest::ProductsWithManagedAssItest.Country", TEST_SCHEMA));
+        assertTrue(HanaITestUtils.checkExistOfPublicSynonym(connection, "itest::ProductsWithManagedAssItest.Country"));
+
+      } finally {
+        HanaITestUtils.dropTable(connection, stmt, "itest::ProductsWithManagedAssItest.Orders", TEST_SCHEMA);
+        HanaITestUtils.dropTable(connection, stmt, "itest::ProductsWithManagedAssItest.Country", TEST_SCHEMA);
+        HanaITestUtils.dropSchema(stmt, TEST_SCHEMA);
+      }
+    }
+  }
+
+  @Test
+  public void testHDBDDWithManagedAssWithUsingOnDiffSchema()
+      throws XSKDataStructuresException, SynchronizationException, IOException, SQLException {
+    try (Connection connection = datasource.getConnection(); Statement stmt = connection.createStatement()) {
+      String schemaName2 = "TEST_SCHEMA 2";
+      try {
+        HanaITestUtils.createSchema(stmt, TEST_SCHEMA);
+        HanaITestUtils.createSchema(stmt, schemaName2);
+
+        LocalResource resource = XSKHDBTestModule.getResources( //
+            "/usr/local/target/dirigible/repository/root", //
+            "/registry/public/itest/ProductsWithManagedAssWithUsingItest.hdbdd", //
+            "/registry/public/itest/ProductsWithManagedAssWithUsingItest.hdbdd" //
+        );
+
+        LocalResource resource2 = XSKHDBTestModule.getResources( //
+            "/usr/local/target/dirigible/repository/root", //
+            "/registry/public/itest/Status.hdbdd", //
+            "/registry/public/itest/Status.hdbdd" //
+        );
+
+        facade.handleResourceSynchronization(resource);
+        facade.handleResourceSynchronization(resource2);
+        facade.updateEntities();
+
+        assertTrue(HanaITestUtils.checkExistOfTable(connection, "itest::ProductsWithManagedAssWithUsingItest.Orders", TEST_SCHEMA));
+        assertTrue(HanaITestUtils.checkExistOfPublicSynonym(connection, "itest::ProductsWithManagedAssWithUsingItest.Orders"));
```

Comment:
can we add message to all asserts - "Expected table was not found", otherwise if the test fail you will see expected true but was false